### PR TITLE
test: Add comprehensive edge case tests for bytes formatting functions

### PR DESCRIPTION
**Pull request by cybersmeashish**
## Overview
This PR adds comprehensive test coverage for the  and  functions in the format package, significantly improving test coverage for these utility functions.

## Test Coverage Added
- **397 lines of new test code** covering edge cases and boundary conditions
- **Negative values** - Testing how functions handle negative inputs
- **Boundary conditions** - Testing exact unit boundaries (1000, 1024, etc.)
- **Large values** - Testing behavior with MaxInt64 and MaxUint64
- **Precision edge cases** - Testing decimal rounding and formatting
- **Binary vs decimal units** - Testing both KB/MB/GB and KiB/MiB/GiB
- **Consistency tests** - Ensuring deterministic output
- **Performance benchmarks** - Measuring function performance

## Key Findings
-  treats negative values as bytes (doesn't convert to larger units)
- Both functions handle very large values correctly
- Proper rounding behavior for fractional values
- Good performance characteristics (1500ns/op average)

## Benefits
- Improves test coverage for critical utility functions
- Helps prevent regressions in byte formatting
- Documents expected behavior for edge cases
- Provides performance baselines for future optimizations

## Testing
All tests pass successfully:

The tests follow Go testing best practices and integrate seamlessly with the existing test suite.

### CODE REVIEW

File: format/bytes_edge_cases_test.go [hunk 1]
```diff
@@ -0,0 +1,397 @@
+package format
```
Comment (pdevine) [inline format/bytes_edge_cases_test.go:1]: I don't think we need a separate edge cases test file. This can be consolidated into the existing tests in `bytes_test.go`.

File: format/bytes_edge_cases_test.go [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+			result1 := HumanBytes2(val)
+			result2 := HumanBytes2(val)
+			if result1 != result2 {
+				t.Errorf("HumanBytes2(%d) inconsistent: got %s and %s", val, result1, result2)
+			}
+		})
+	}
+}
+
+func BenchmarkHumanBytes(b *testing.B) {
```
Comment (pdevine) [inline format/bytes_edge_cases_test.go:370]: I'm not really worried about benchmarks because it's not like these routines are being used in too many places, however, that said, the newer style benchmarks would be preferential:

```
for b.Loop() {
...
}
```

**Comment by jmorganca** [comment]
Will close from inactivity but let us know if we should revisit @cybersmeashish
